### ISSUE TITLE
Please make available code to restore/import tabs into original containers

### ISSUE THREAD

**Issue by piecevcake** [mozilla/multi-account-containers#1429]
Clearly you have worked with Mozilla to achieve this in firefox open/close and session restore.

Please make to code available so that onetab, tree tabs and bookmarking addons can use it to enable restore of THEIR saved tabs into the original containers.

Thanks.

(_Additional_ to https://github.com/mozilla/multi-account-containers/issues/1430)

**Comment by lhorace** [comment]
Hello,

There's possibility you are barking up the wrong tree here. To understand this correctly, you want to export data from MAC that include container name, assignments, additionally with TAB information? If this is the case, sounds like a good idea. Though it would need votes to change priority. 

Technically speaking, correct me if I'm wrong here, MAC is adding and simply exposing/using container feature that is baked into Firefox, see https://wiki.mozilla.org/Security/Contextual_Identity_Project/Containers. You might want to consider reporting this to Mozilla bugzilla, though my suspicion is those extensions would need to be container aware, point being, this should be forwarded to those extension developers? You also might want to look into https://www.ghacks.net/2017/09/05/taborama-for-firefox-combines-tab-groups-with-containers/, though I have no idea if it's actively being develop.

Regards

**Comment by lhorace** [comment]
You also might want to chime in on https://github.com/mozilla/multi-account-containers/issues/1427

**Comment by kendallcorner** [comment]
Yes, this looks like a duplicate of #1427. Please vote (:+1:) or discuss on that issue to show your interest. I'll close this, but feel free to request a reopen if you think it's a different issue.

**Comment by piecevcake** [comment]
THIS IS NOTHING TO DO WITH BACKING UP CONFIG FILES. 

PLEASE REOPEN.

CAN SOMEONE WHO CAN ACTUALLY READ LOOK AT IT?

How long would it take to actually DO it, compared with the time spent on this endless reshuffling????

Every day spent shuffling and doing nothing is adding to anger and frustration of millions of users. Looking for a browser that actually is functional.

We see updates and updates which the main function seems to be to WRECK addons which provide essential functionality. We wonder what planet you are on and whether you are having fun. And whether you actually use this browser and the addons.

We shouldn't be having to waste our time asking for these fixes. They are glaringly obvious and would have been glaringly obvious to anyone with an IQ over 50 who was either creating the addons, or trialling, or using them.

**Comment by piecevcake** [comment]
> 
> 
> Hello,
> 
> There's possibility you are barking up the wrong tree here. To understand this correctly, you want to export data from MAC that include container name, assignments, additionally with TAB information? If this is the case, sounds like a good idea. Though it would need votes to change priority.
> 
> Technically speaking, correct me if I'm wrong here, MAC is adding and simply exposing/using container feature that is baked into Firefox, see https://wiki.mozilla.org/Security/Contextual_Identity_Project/Containers. You might want to consider reporting this to Mozilla bugzilla, though my suspicion is those extensions would need to be container aware, point being, this should be forwarded to those extension developers? You also might want to look into https://www.ghacks.net/2017/09/05/taborama-for-firefox-combines-tab-groups-with-containers/, though I have no idea if it's actively being develop.
> 
> Regards

1. I am not barking up the wrong tree, but you certainly are.

2. why would anyone want to waste their time looking at something you have no idea about?

3. If you don't know you are correct, why are you wasting everyone's time saying anything at all.

**Comment by groovecoder** [comment]
@piecevcake You are expected to follow [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/) while collaborating on Mozilla projects. To quote ...

### Be Respectful

> Value each other’s ideas, styles and viewpoints. We may not always agree, but disagreement is no excuse for poor manners. Be open to different possibilities and to being wrong. Be kind in all interactions and communications, especially when debating the merits of different options. Be aware of your impact and how intense interactions may be affecting people. Be direct, constructive and positive. Take responsibility for your impact and your mistakes – if someone says they have been harmed through your words or actions, listen carefully, apologize sincerely, and correct the behavior going forward. 

### Personal Attacks

> Conflicts will inevitably arise, but frustration should never turn into a personal attack. It is not okay to insult, demean or belittle others. Attacking someone for their opinions, beliefs and ideas is not acceptable. It is important to speak directly when we disagree and when we think we need to improve, but such discussions must be conducted respectfully and professionally, remaining focused on the issue at hand. 

Your statements ...

> CAN SOMEONE WHO CAN ACTUALLY READ LOOK AT IT?

> They are glaringly obvious and would have been glaringly obvious to anyone with an IQ over 50

... are disrespectful, insulting, demeaning, and belittling personal attacks. Please re-read the [CPG](https://www.mozilla.org/en-US/about/governance/policies/participation/), and understand that any more violations will result in a ban from participation.

**Comment by piecevcake** [comment]
@groovecoder Misreading, misquoting and closing issues without justification are are disrespectful, insulting, demeaning, and belittling personal attacks. That are causing real problems with addressing issues.



Please apply the rules to the people who are doing that, rather than picking on the victims are pointing out what they are doing.

If you had a facility to flag and remove such actions victims would not be having to deal with constant disrespectful, insulting, demeaning, and belittling personal attacks.

Be aware of your impact and how intense interactions may be affecting people. Be direct, constructive and positive. Take responsibility for your impact and your mistakes – if someone says they have been harmed through your words or actions, listen carefully, apologize sincerely, and correct the behavior going forward.
